src/main/java/com/artipie/rpm/misc/StorageLock.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.rpm.misc;
+
+import com.artipie.asto.Key;
+import com.artipie.asto.Storage;
+import com.artipie.asto.blocking.BlockingStorage;
+import java.util.UUID;
+import java.util.function.Predicate;
+
+/**
+ * Locker for the storage.
+ * @since 0.9
+ * @todo #230:30min Use this class in RPM update to prohibit parallel update of the same repository:
+ *  add lock before starting the update and release it on terminate.
+ */
+public final class StorageLock {
+
+    /**
+     * Lock file name.
+     */
+    private static final String FNAME = "lock-%s";
+
+    /**
+     * Lock file name pattern.
+     */
+    private static final String PTRN = "%s/lock-[a-z0-9-]{36}";
+
+    /**
+     * Storage.
+     */
+    private final Storage storage;
+
+    /**
+     * Repository key.
+     */
+    private final Key repo;
+
+    /**
+     * Lock file key.
+     */
+    private final Key file;
+
+    /**
+     * Ctor.
+     * @param storage Storage
+     * @param repo Repo key
+     */
+    public StorageLock(final Storage storage, final Key repo) {
+        this.storage = storage;
+        this.repo = repo;
+        this.file = new Key.From(
+            this.repo, String.format(StorageLock.FNAME, UUID.randomUUID().toString())
+        );
+    }
+
+    /**
+     * Locks storage by adding random file to it.
+     * @throws IllegalStateException On error and if storage is already locked
+     */
+    void lock() {
+        final BlockingStorage bsto = new BlockingStorage(this.storage);
+        try {
+            if (this.noLocks(bsto)) {
+                bsto.save(this.file, new byte[]{});
+                if (bsto.list(this.repo).stream().filter(this.lockKeyPredicate()).count() > 1) {
```

Comment:
@olenagerasimova since this is very similar to `noLocks`, can(y we have a method returning `bsto.list(this.repo).stream().filter(this.lockKeyPredicate()).count()` instead and use it in both if to check that it's equal to zero or when it is not > 0. This would be clearer I think, also this would allow to inline the predicate directly in the method.